### PR TITLE
local Storage

### PR DESCRIPTION
**Pull request by ZabihullahNooriWardak**
in this branch  we worked on local storage for form text fields .

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -243,3 +243,47 @@ ${dynamicImage}
...
+  };
+  try {
+    const forLocalStorage = JSON.stringify(inputFieldObject);
+    window.localStorage.setItem('userData', forLocalStorage);
+  } catch (e) {
+    console.log('error occured while storing data to localStorage');
+  }
+}
+
+// console.log(localStorage.getItem('userData'));
```
Comment (abdulhamiid) [inline script.js]: To follow our [list of best practices](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md), please do not commit old pieces of code as inline comments. They will make your project look messy. If you need to review a previous version of your code, you can always use git history. 🤗

File: script.js [hunk 2]
```diff
@@ -243,3 +243,47 @@ ${dynamicImage}
...
+  const userEmail = document.getElementById('email');
+  const userComment = document.getElementById('textArea');
+  const uname = userName.value;
+  const email = userEmail.value;
+  const comment = userComment.value;
+  const inputFieldObject = {
+    name: uname,
+    email,
+    comment,
+  };
```
Comment (abdulhamiid) [inline script.js]: According to the [project requirements](https://github.com/microverseinc/curriculum-html-css/blob/main/portfolio/10_localstorage.md), please let's implement the following interactions and ensure that everything works fine. 💪

image

Currently, your app is not working as expected, and the data from the `form` are not saved to the local storage. 👀

Error in the console
image

Source
image

**Review by abdulhamiid — CHANGES_REQUESTED** [review]
# Status: Required Changes :recycle:
Hi Team 👋, 
Alt Text
Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Required Changes ♻️
_Please kindly check the comments under the review._

- Your commits


image

[REQUIRED] - Please always use descriptive commit messages - each commit should be small enough to describe it in one sentence. 

The commit message should be written in the `imperative tense` and `capitalized`. [You can read more about how to write descriptive commit messages](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8).

- Here are some examples of descriptive commits:

image

🏮 As it is not easy to change commit history, please correct this behavior in the future. 👍

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: script.js [hunk 3]
```diff
@@ -243,3 +243,45 @@ ${dynamicImage}
...
```
Comment (Qoosim) [inline script.js]: - [ ] Great work implementing required changes from the previous reviewer. However, there's one more thing you guys need to work on before your project could be approved. It is required that when the When the user loads the page, the code checks if there's any saved data in the local storage and if so the input fields are pre-filled with the data. What this means is that, if the user enters data into the form without submitting the form, when the user reload the page, the user's data should remain on the form. Kindly check below code for insight :point_down: 

```
    window.onload = () => {
      let savedFormData = localStorage.getItem('contactFormInput');
      savedFormData = JSON.parse(savedFormData);
      // Check if the form data object is found on localStorage
      if (savedFormData) {
        // populate inputs values if data was found
        // ex: nameInput.value = savedFormData.name
       }
    };

```

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle: 

Hi @team, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


## Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct GitHub Flow. :heavy_check_mark: 



 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: script.js [hunk 4]
```diff
@@ -243,3 +243,49 @@ ${dynamicImage}
...
```
Comment (skyv26) [inline script.js]: - [ ] Well done 👏🏻 , you did an awesome job by implementing the changes as per the previous code reviewer's request 👍🏻 but you guys still not following the things properly, you need to work on it before your project could be approved. It is required that when the user loads the page, the code checks if there are any saved data in the local storage and if so the input fields are pre-filled with the data. What this means is that, if the user enters data into the form without submitting the form, when the user reloads the page, the user's data should remain on the form. Kindly check the below code for insight 👇

```javascript
    window.onload = () => {
      let savedFormData = localStorage.getItem('contactFormInput');
      savedFormData = JSON.parse(savedFormData);
      // Check if the form data object is found on localStorage
      if (savedFormData) {
        // populate inputs values if data was found
        // ex: nameInput.value = savedFormData.name
       }
    };

```

### Video Demo For Understanding
[Screencast from 01-07-23 09:24:27 AM IST.webm](https://github.com/ZabihullahNooriWardak/Zabih-portfolio/assets/66476329/0e3904ac-fbf7-4ad5-9e1d-a422455d9567)

### Please resolve the below console error
Screenshot from 2023-07-01 09-22-05

**Review by skyv26 — CHANGES_REQUESTED** [review]
## Hi @ZabihullahNooriWardak, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## STATUS: REQUIRE CHANGES 🔴 



## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Implement the features ✅ 
- Professional `README` template ✅ 
- Linters are green ✅ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

#### Cheers, and Happy coding!👏👏👏

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by usorfaitheloho — APPROVED** [review]
Hi @ZabihullahNooriWardak 👋,

# STATUS: APPROVED 🕺🥳 

Congratulations🍾🥳!! Your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 



Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**